src/device/log.cpp

```diff
@@ -0,0 +1,136 @@
+#include "misc_log_ex.h"
+#include "log.hpp"
+
+namespace hw {
+    namespace ledger {
+
+    void buffer_to_str(char *to,  const char *buff, int len) {
+      for (int i=0; i<len; i++) {
+        sprintf(to+2*i, "%.02x", (unsigned char)buff[i]);
+      }
+    }
+    void bufferLE_to_str(char *to,  const char *buff, int len) {
+      for (int i=0; i<len; i++) {
+        sprintf(to+2*i, "%.02x", (unsigned char)buff[31-i]);
+      }
+    }
+
+    void log_hexbuffer(std::string msg,  const char* buff, int len) {
+      char logstr[1024];
+      buffer_to_str(logstr,  buff, len);
+      MCDEBUG("ledger", msg<< ": " << logstr);
+    }
+
+    void log_hexbufferLE(std::string msg,  const char* buff, int len) {
+      char logstr[1024];
+      bufferLE_to_str(logstr,  buff, len);
+      MCDEBUG("ledger", msg<< ": " << logstr);
+    }
+
+    void log_message(std::string msg,  std::string info ) {
+      MCDEBUG("ledger", msg << ": " << info);
+    }
```

Comment:
Move of the above is not necessary, just the endian specific stuff I think.

Reply:
cleanup done